### ISSUE TITLE
System.Text.Json fails to serialize and throws internal System.NullReferenceException

### ISSUE THREAD

**Issue by Eneuman** [dotnet/runtime#79978]
### Description

When using a custom converters System.Text.Json failes to serialize and throws the following internal exception:

"Object reference not set to an instance of an object."

```
   at System.Text.Json.Serialization.JsonDictionaryConverter`3.OnTryWrite(Utf8JsonWriter writer, TDictionary dictionary, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.GetMemberAndWriteJson(Object obj, WriteStack& state, Utf8JsonWriter writer)
   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryWrite(Utf8JsonWriter writer, T value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.WriteCore(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonSerializer.WriteUsingSerializer[TValue](Utf8JsonWriter writer, TValue& value, JsonTypeInfo jsonTypeInfo)
   at System.Text.Json.JsonSerializer.WriteStringUsingSerializer[TValue](TValue& value, JsonTypeInfo jsonTypeInfo)
   at System.Text.Json.JsonSerializer.Serialize[TValue](TValue value, JsonSerializerOptions options)
   at Program.<Main>$(String[] args) in C:\Users\PerBornsjö\source\repos\ConsoleApp3\ConsoleApp3\Program.cs:line 11

```

### Reproduction Steps

Open VS and create a new concole application using .Net 6

Paste the following code:

```
using System.Text.Json;
using System.Text.Json.Serialization;

var serializerTest = new SerializerTest();
serializerTest.LoggingProperties["Test1"] = "10";

var jsonSerializerOptions = new JsonSerializerOptions();
jsonSerializerOptions.Converters.Add(new DictionaryStringObjectJsonConverter());

// This lines failes
var json = JsonSerializer.Serialize(serializerTest, jsonSerializerOptions);

public class SerializerTest
{
    public IDictionary<string, object> LoggingProperties { get; private set; } = new Dictionary<string, object>();
}

public class DictionaryStringObjectJsonConverter : JsonConverter<Dictionary<string, object>>
{
    public override bool CanConvert(Type typeToConvert)
    {
        return typeToConvert == typeof(Dictionary<string, object>);
    }

    public override Dictionary<string, object> Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
    {
        var dictionary = new Dictionary<string, object>();
        return dictionary;
    }

    public override void Write(Utf8JsonWriter writer, Dictionary<string, object> value, JsonSerializerOptions options)
    {
    }
}
```

Run it and the error will occure.
This works fine in .Net 7 but failes in latest .Net 6.

### Expected behavior

The object serializes correctly.

### Actual behavior

A null referense exception is thrown.

### Known Workarounds

Upgrading to .Net 7 will solve the problem.

### Configuration

.Net 6.0.12
Windows x64

**Comment by danmoseley** [comment]
Is upgrading to .NET 7 an option for you?

**Comment by Eneuman** [comment]
@danmoseley Unfortunately not :( 

I'm working on a PR for moving away from Newtonsoft to System.Text.Json in the project Azure Bridge to Kubernetes and there are alot of dictionary that's beeing serialized. They just move to .Net 6 and I dont know if they want to move to .Net 7 just yet but I can check.

**Comment by eiriktsarpalis** [comment]
It should be possible to reference the v7 System.Text.Json NuGet package from .NET 6 without upgrading.

**Comment by eiriktsarpalis** [comment]
This was fixed in #63686. Unfortunately, it doesn't meet the bar for a backporting fix to .NET 6. Recommended action is either upgrading to .NET 7 or referencing the latest System.Text.Json nuget package release.
